core/src/main/java/org/apache/calcite/tools/RelBuilder.java

```diff
@@ -1616,6 +1619,58 @@ public RelBuilder minus(boolean all, int n) {
     return setOp(all, SqlKind.EXCEPT, n);
   }
 
+  /**
+   * Creates a {@link LogicalDeltaTableScan} of a delta work table (used to accumulate results in
+   * recursive union operation), using as its row type the top of the stack.
+   * Returns this builder.
+   */
+  public RelBuilder deltaScan(String tableName) {
```

Comment:
Since the notion of Delta already exists for the case of streams I would suggest to use the notion of transient tables for the new operators. For example deltaScan becomes transientScan, LogicalDeltaTableScan becomes LogicalTransientTableScan etc.

Reply:
+1
Delta would indeed confuse.
Especially for `LogicalDeltaDeltaScan`

Reply:
Agree, *delta* renamed as *transient*